### PR TITLE
fix: Show online or mixed location in event page header

### LINKED ISSUE #5161: Public Event Page: Display Info for online, mixed etc. on header
The header of events is currently only showing info related to locations. When the organizer chooses online, mixed, or to be announced the header shows a location icon but nothing else. Therefore please implement the following on the public event page header, when the organizer chooses:
* Online Event show "Online Event"
* Mixed Event show the info "In-Person Event and Online Event" + show the address of the event
* To be announced show "Location to be announced" 

Screenshot from 2020-09-27 14-04-42

### PR DESCRIPTION
**Pull request by HADES-01**
Fixes #5161 

#### Short description of what this resolves:
Fixed the issue of empty location line in the header of the public event page.

#### Changes proposed in this pull request:
- Fixed by creating a computed property as suggested by @iamareebjamal

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes

### CODE REVIEW

File: app/models/event.js [hunk 1]
```diff
@@ -181,6 +181,21 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  headererLocation: computed('locationName', 'online', function() {
+    if (this.locationName && this.online) {
+      return `In-Person Event and Online Event ${this.locationName}`;
+    }
```
Comment (niranjan94) [inline app/models/event.js:189]: Codacy found an issue: [Closing curly brace does not appear on the same line as the subsequent block.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6259866)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/models/event.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6259866&bid=20382915)

File: app/models/event.js [hunk 2]
```diff
@@ -181,6 +181,21 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
...
+    else if (!this.locationName && this.online) {
```
Comment (iamareebjamal) [inline app/models/event.js:190]:
```suggestion
    else if (this.online) {
```

File: app/models/event.js [hunk 3]
```diff
@@ -181,6 +181,21 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
...
+      return 'Online Event';
+    }
+    else if (!this.online && this.locationName) {
```
Comment (iamareebjamal) [inline app/models/event.js:193]:
```suggestion
    else if (this.locationName) {
```

File: app/models/event.js [hunk 4]
```diff
@@ -181,6 +181,21 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  headererLocation: computed('locationName', 'online', function() {
```
Comment (iamareebjamal) [inline app/models/event.js:186]:
```suggestion
  headerLocation: computed('locationName', 'online', function() {
```

**Comment by iamareebjamal** [comment]
Use correct formatting

File: app/models/event.js [hunk 5]
```diff
@@ -181,6 +181,18 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  headerLocation: computed('locationName', 'online', function() {
+    if (this.locationName && this.online) {
+      return `In-Person Event and Online Event ${this.locationName}`;
+    } else if (!this.locationName && this.online) {
```
Comment (iamareebjamal) [inline app/models/event.js:189]:
```suggestion
    } else if (this.online) {
```

File: app/models/event.js [hunk 6]
```diff
@@ -181,6 +181,18 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
...
+      return 'Online Event';
+    } else if (!this.online && this.locationName) {
```
Comment (iamareebjamal) [inline app/models/event.js:191]:
```suggestion
    } else if (this.locationName) {
```

File: app/models/event.js [hunk 7]
```diff
@@ -181,6 +181,18 @@ export default class Event extends ModelBase.extend(CustomPrimaryKeyMixin, {
 
   sessionsByState: computed('sessions', function() {
     return groupBy(this.sessions.toArray(), 'data.state');
+  }),
+
+  headerLocation: computed('locationName', 'online', function() {
```
Comment (iamareebjamal) [inline app/models/event.js:186]: Move this to the controller rather than the model

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/models/event.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6259866&bid=20382915)

**Review by iamareebjamal — APPROVED** [review]

File: app/templates/public.hbs [hunk 8]
```diff
@@ -14,7 +14,7 @@
               <h5 class="event time ends">{{t 'To'}} {{header-date this.model.endsAt this.model.timezone}}</h5>
             {{/if}}
             <h1 class="event name">{{this.model.name}}</h1>
-            <h4 class="event location"><i class="icon map marker alternate"></i>{{this.model.locationName}}</h4>
+            <h4 class="event location"><i class="icon map marker alternate"></i>{{this.model.headerLocation}}</h4>
```
Comment (iamareebjamal) [inline app/templates/public.hbs:17]:
```suggestion
            <h4 class="event location"><i class="icon map marker alternate"></i>{{this.headerLocation}}</h4>
```

**Review by iamareebjamal — CHANGES_REQUESTED** [review]
Wrong reference

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6259866&bid=20382915)

File: app/controllers/public.js [hunk 9]
```diff
@@ -20,6 +20,19 @@ export default class PublicController extends Controller {
...
+      return `In-Person Event and Online Event ${this.model.locationName}`;
+    } else if (this.model.online) {
+      return 'Online Event';
+    } else if (this.model.locationName) {
+      return this.model.locationName;
+    } else {
+      return 'Location to be announced';
+    }
+  }
+
```
Comment (niranjan94) [inline app/controllers/public.js:35]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6259866)

File: app/templates/public.hbs (hunk 8 above)
Comment (iamareebjamal) [reply app/templates/public.hbs:17]: Do not resolve conversation. It is resolved automatically once suggested changes are made

File: app/controllers/public.js [hunk 10]
```diff
@@ -20,6 +20,19 @@ export default class PublicController extends Controller {
     return this.session.currentRouteName && this.session.currentRouteName !== 'public.cfs.new-session' && this.session.currentRouteName !== 'public.cfs.new-speaker' && this.session.currentRouteName !== 'public.cfs.edit-speaker' && this.session.currentRouteName !== 'public.cfs.edit-session';
   }
 
+  @computed('locationName', 'online')
```
Comment (iamareebjamal) [inline app/controllers/public.js:23]: Wrong computed properties

Every event is shown as Online Event. Please test the PR properly

Comment (HADES-01) [reply app/controllers/public.js:23]: @iamareebjamal It shows online event at first, but when you refresh it shows the correct value. Then this value shows up for the next event you click. Can you help me, I am not able to figure it out myself.

Comment (iamareebjamal) [reply app/controllers/public.js:23]: `@computed('model.locationName', 'model.online')`

Comment (HADES-01) [reply app/controllers/public.js:23]: @iamareebjamal  Thanks

**Pull request merged**